barretenberg/acir_tests/sol-test/src/index.js

```diff
@@ -0,0 +1,177 @@
+const bytecode = contract.evm.bytecode.object;
+const abi = contract.abi;
+
+const launchAnvil = async (port) => {
+  const handle = spawn("anvil", ["-p", port]);
+
+  // wait until the anvil instance is ready on port 8545
+  await new Promise((resolve, reject) => {
+    // If we get an error reject, which will cause the caller to retry on a new port
+    handle.stderr.on("data", (data) => {
+      const str = data.toString();
+      if (str.includes("error binding")) {
+        reject("we go again baby")
+      }
+    });
+
+    // If we get a success resolve, anvil is ready
+    handle.stdout.on("data", (data) => {
+      const str = data.toString();
+      if (str.includes("Listening on")) {
+        resolve(undefined);
+      }
+    });
+  });
+
+  return handle;
+}
+
+const deploy = async (signer) => {
+    const factory = new ethers.ContractFactory(abi, bytecode, signer);
+    const deployment = await factory.deploy();
+    const deployed = await deployment.waitForDeployment();
+    return await deployed.getAddress();
+}
+
+/**
+ * 
+ * @param {number} numPublicInputs 
+ * @param {Array<String>} proofAsFields 
+ * @returns {Array<Tuple<Array<String>,number>}
+ */
+const readPublicInputs = (proofAsFields) => {
+  const publicInputs = [];
+  // A proof with no public inputs is 93 fields long
+  const numPublicInputs = proofAsFields.length - 93;
+  for (let i = 0; i < numPublicInputs; i++) {
+    publicInputs.push(proofAsFields[i]);
+  }
+  return [numPublicInputs, publicInputs];
+}
+
+// start anvil
+const getAnvil = async () => {
+  const port = Math.floor(Math.random() * 10000) + 10000;
+  try {
+    return [await launchAnvil(port), port];
+  } catch (e) {
+    // Recursive call should try again on a new port in the rare case the port is already taken
+    // yes this looks dangerous, but it relies on 0-10000 being hard to collide on
+    return getAnvil();
+  }
+}
+
+const [anvil, randomPort] = await getAnvil();
+const killAnvil = () => {
+  anvil.kill();
+  console.log(testName, " complete")
+}
+
+try {
+  const proofAsFieldsPath = getEnvVar("PROOF_AS_FIELDS");
+  const proofAsFields = readFileSync(proofAsFieldsPath);
+  const [numPublicInputs, publicInputs] = readPublicInputs(JSON.parse(proofAsFields.toString()));
+
+  const proofPath = getEnvVar("PROOF");
+  const proof = readFileSync(proofPath);
+  
+  // Cut the number of public inputs off of the proof string
+  const proofStr = `0x${proof.toString("hex").substring(64*numPublicInputs)}`;
+
+  // Get the contract artifact
+  const key = "0xac0974bec39a17e36ba4a6b4d238ff944bacb478cbed5efcae784d7bf4f2ff80";
+  const provider = new ethers.JsonRpcProvider(`http://localhost:${randomPort}`);
+  const signer = new ethers.Wallet(key, provider);
+
+  // deploy 
+  const address = await deploy(signer);
+  const contract = new ethers.Contract(address, abi, signer);
+
+  // Run the test
+  const result = await contract.test(proofStr, publicInputs);
```

Comment:
Calling verify directly should do the job, I will probably not run this in CI since we are running all commands, but it is useful to have this in the future. 
The purpose of making this was to check FOR SURE it wasnt solidity causing the issues